### PR TITLE
test: SANDBOX-629: refactor CreateSpaceRequest

### PR DESCRIPTION
**Pull request by rsoaresd**
# Description
Remove unnecessary standalone space creation. For more info, check PR #983 or [SANDBOX-629](https://issues.redhat.com/browse/SANDBOX-629).

## Issue ticket number and link
[SANDBOX-629](https://issues.redhat.com/browse/SANDBOX-629)

### CODE REVIEW

**Review by MatousJobanek — APPROVED** [review]
Nice :+1:

File: testsupport/space/spacerequest.go [hunk 1]
```diff
@@ -55,10 +55,14 @@ func WithName(name string) SpaceRequestOption {
 }
 
 func CreateSpaceRequest(t *testing.T, awaitilities wait.Awaitilities, memberName string, opts ...SpaceRequestOption) (*toolchainv1alpha1.SpaceRequest, *toolchainv1alpha1.Space) {
+	hostAwait := awaitilities.Host()
 	memberAwait, err := awaitilities.Member(memberName)
 	require.NoError(t, err)
 	// let's first create a parentSpace
-	parentSpace, _, _ := CreateSpace(t, awaitilities, testspace.WithTierName("appstudio"), testspace.WithSpecTargetCluster(memberAwait.ClusterName))
+	_, mur := testsupport.CreateUserSignupWithSpaceTier(t, awaitilities, "appstudio")
```
Comment (MatousJobanek) [inline testsupport/space/spacerequest.go:62]: one thing I just realized - the original code provisioned the space in the given member cluster, the rest of the tests relies on this fact, but the new code would pick any of the member clusters that has enough capacity.
This should be changed so the space is always provisioned to the given member cluster.

**Review by MatousJobanek — CHANGES_REQUESTED** [review]

File: testsupport/signup_request.go [hunk 2]
```diff
@@ -353,3 +365,16 @@ func Close(t *testing.T, resp *http.Response) {
...
+func CreateUserSignupWithSpaceTier(t *testing.T, awaitilities wait.Awaitilities, tier string) (*toolchainv1alpha1.UserSignup, *toolchainv1alpha1.MasterUserRecord) {
+	userSignup, mur := NewSignupRequest(awaitilities).
+		ManuallyApprove().
+		RequireConditions(wait.ConditionSet(wait.Default(), wait.ApprovedByAdmin())...).
+		SpaceTier(tier).
+		EnsureMUR().
+		Execute(t).
+		Resources()
+
+	return userSignup, mur
```
Comment (MatousJobanek) [inline testsupport/signup_request.go:379]: let's also wait for the space and return it from the function so there is some added value in the function and it doesn't only abstract the fluent-api.
```go
	space, err := awaitilities.Host().WaitForSpace(t, mur.Name)
	require.NoError(t, err)
	return userSignup, mur, space
```

Comment (MatousJobanek) [reply testsupport/signup_request.go:379]: or you could update the SignupRequest API so the `Resources()` function returns also the Space object (if provisioned)

Comment (MatousJobanek) [reply testsupport/signup_request.go:379]: Updating returned values of the `Resources()` function of the SignupRequest API would cause many changes across the whole e2e project, so if you decide to do that, then let's do it in a separate PR.

Comment (rsoaresd) [reply testsupport/signup_request.go:379]: I am more inclined to the first option. WDYT?

File: testsupport/space/spacerequest.go (hunk 1 above)
Comment (rsoaresd) [reply testsupport/space/spacerequest.go:62]: Good catch!! My bad, sorry! I will address it!

File: testsupport/signup_request.go (hunk 2 above)
Comment (rsoaresd) [reply testsupport/signup_request.go:380]: I think I will go with option 2, after all 😄 

Regarding CreateUserSignupWithSpaceTier only abstract the fluent-api, I create this function to avoid duplicated code, since we will need it in other tests like PR #984, PR #985, and PR #990

Comment (MatousJobanek) [reply testsupport/signup_request.go:380]: yeah, option 2 is a bit more work, but it will help us with cleaning up some of the duplicated code (retrieval of the Spaces). So I'm for it :+1: 

> Regarding CreateUserSignupWithSpaceTier only abstract the fluent-api, I create this function to avoid duplicated code

I understand that there can be some duplication, but this is something (with the number of test-cases) we just cannot avoid. Also, I have a feeling that the number of helper functions is growing and it's sometimes hard to follow which functions does what and why, so if it was there only to abstract the fluent API, then I would rather drop it and go with the small duplication of the same. In addition to that, it's sometimes better to call the fluent API with the exact values so it's clear what is set and used - like the target member cluster the user is gonna be provisioned to.

**Review by alexeykazakov — APPROVED** [review]
Thanks!

**Review by mfrancisc — APPROVED** [review]
Nice job 👍

**Review by MatousJobanek — APPROVED** [review]
